### ISSUE TITLE
OneDrive fail to recognize existing files due to uppercase/lowercase discrepancy

### ISSUE THREAD

**Issue by yonjah** [rclone/rclone#1088]
rclone v1.35-52-g7470255 on Linux (Fedora 23) 64 bit OneDrive
rclone copy/change
sample output of change - 
```
...
2017/02/01 20:44:16 MVI_0512.MP4: File not in Local file system at /home/user/Pictures
...
2017/02/01 20:44:16 MVI_0512.mp4: File not in One drive root 'backup/Private/Pictures'
...
```

I think this been hapening on older versions as well but it took me time to pinpoint the issue
it seem like I have some files that their extension changed from being lowercased to capitalized.
I'm not sure if this was some bug that was fixed in previous issue of rclone or a bug that existed in OneDrive and was fixed but the fact is I have some file on the remote with mismatching file names 
for example -
some_video.mp4 (local)
some_video.MP4 (remote)

when rclone copy runs it allways uploads `some_video.mp4` since it does not exist on the remote
but I guess OneDrive recognizes the `some_video.MP4` exists and just keeps it as is (with the old name) so next time copy will run it will still think `some_video.mp4` is missing and it will upload it again.

I can manually solve the issue by removing `some_video.MP4` and upload it again, but I have quite a lot of this files and I'm not sure if this won't happen again.

It would be great if rclone could detect casing discrepancy without uploading and just fix the file name (according to the copy direction).
If this is not possible either making sure file name is correct after upload and if not fix it or just ignoring casing all together with one drive would be a lot better.

**Comment by ncw** [comment]
>  I have some files that their extension changed from being lowercased to capitalized.
I'm not sure if this was some bug that was fixed in previous issue of rclone or a bug that existed in OneDrive and was fixed

rclone expects that if it uploads a file in a particular case, then the remote will keep that case.  I don't think that is an unreasonable expectation even for case insensitive remotes like OneDrive.

According to some brief tests onedrive still behaves like that - if I upload FILE.JPG it keeps it like that in the listings and doesn't change it to FILE.jpg

So is it possible that these got renamed locally?

You can try a latest beta with --track-renames if you want the detecting files have moved feature!

**Comment by yonjah** [comment]
I'll try to see if I can find more info what caused the renames if they'll happen again, but files never changed locally and I only use rclone for the uploads

I tried it with track renamed unfortunately that didn't really worked.
I  ran 
```shell
rclone copy --track-renames path remote:path
```
That seemed to be working and the log indicated many files that were renamed
from ####.MP4 to ####.mp4 unfortunately after every move rclone also indecated
deleting the original file (####.MP4) and because OneDrive is case insensitive it just deleted the file.
I'm not sure if the actual rename succeed in changing the name to lowercase before it was deleted or not since this happened almost instantly.

Running another copy uploaded the missing files (this time with the correct name) so I guess if I really want to solve this issue I can run a `copy` command with `--track-renames` on all my backup deleting all files that have the wrong name and then run it again to reupload all the missing files but this sounds a bit risky and it will probably take ages do to my bad internet connection.

My situation is probably a very rare edge case to have a better solution for it. but I think rclone should be aware if the remote is case insensitive and try to avoid removing / overwriting the wrong files.
Can rclone run on windows ? not sure what's going with windows maybe it should also do it with the local fs.

**Comment by ncw** [comment]
> That seemed to be working and the log indicated many files that were renamed
from ####.MP4 to ####.mp4 unfortunately after every move rclone also indecated
deleting the original file (####.MP4) and because OneDrive is case insensitive it just deleted the file.
I'm not sure if the actual rename succeed in changing the name to lowercase before it was deleted or not since this happened almost instantly.

Foo, that is annoying and unexpected! I made an issue to fix it #1094

> Can rclone run on windows ? not sure what's going with windows maybe it should also do it with the local fs.

Yes and agreed.

**Comment by ncw** [comment]
I think this issue is now fixed.
